File: baekjoon/1037/1037.py
'''
약수 python 풀이
https://doctcoder.tistory.com/entry/%EC%95%BD%EC%88%98-%ED%92%80%EC%9D%B4
'''
n=int(input())
arr=list(map(int,input().split()))
arr.sort()
print(arr[0]*arr[-1])

File: baekjoon/1037/1037.cpp
/*
약수 C++ 풀이
https://doctcoder.tistory.com/entry/%EC%95%BD%EC%88%98-%ED%92%80%EC%9D%B4
*/
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main()
{
	int n;
	cin>>n;
	vector<int> v(n);
	for(int i=0;i<n;i++) cin>>v[i];
	sort(v.begin(),v.end());
	cout<<v[0]*v[n-1];
}
